webpack.config.js

```diff
@@ -1,120 +1,20 @@
+    "pretty-print-worker": path.join(projectPath, "utils/pretty-print-worker.js")
   },
-  devtool: "source-map",
+
   output: {
     path: path.join(__dirname, "public/build"),
     filename: "[name].js",
     publicPath: "/public/build"
-  },
-  resolve: {
-    alias: {
-      "devtools/client/shared/vendor/react": "react",
-      "devtools/client/shared/vendor/react-dom": "react-dom",
-      "devtools/shared/event-emitter":
-                    path.join(__dirname, "./public/js/utils/event-emitter"),
-      "devtools": path.join(__dirname, "./public/js/lib/devtools"),
-      "devtools-sham": path.join(__dirname, "./public/js/lib/devtools-sham"),
-      "sdk": path.join(__dirname, "./public/js/lib/devtools-sham/sdk"),
-      "Services": path.join(
-        __dirname, "./public/js/lib/devtools/client/shared/shim/Services"),
-      "networkRequest": path.join(__dirname, "./public/js/utils/networkRequest")
-    }
-  },
-  module: {
-    loaders: [
-      { test: /\.json$/,
-        loader: "json" },
-      { test: /\.js$/,
-        exclude: /(node_modules|bower_components)/,
-        loaders: [
-          "babel?" +
-            defaultBabelPlugins.map(p => "plugins[]=" + p) +
-            "&ignore=public/js/lib"
-        ],
-        isJavaScriptLoader: true
-      },
-      { test: /\.svg$/,
-        loader: "svg-inline" }
-    ]
-  },
-  plugins: [
-    new webpack.DefinePlugin({
-      "process.env": {
-        NODE_ENV: JSON.stringify(NODE_ENV),
-        TARGET: JSON.stringify("local")
-      },
-      "DebuggerConfig": JSON.stringify(getConfig())
-    })
-  ]
-};
-
-if (isDevelopment()) {
-  webpackConfig.module.loaders.push({
-    test: /\.css$/,
-    loader: "style!css"
-  });
-
-  if (getValue("hotReloading")) {
-    webpackConfig.entry.bundle.push("webpack-hot-middleware/client");
-
-    webpackConfig.plugins = webpackConfig.plugins.concat([
-      new webpack.HotModuleReplacementPlugin(),
-      new webpack.NoErrorsPlugin()
-    ]);
-
-    webpackConfig.module.loaders.forEach(spec => {
-      if (spec.isJavaScriptLoader) {
-        spec.loaders.unshift("react-hot");
-      }
-    });
   }
-} else {
-  // Extract CSS into a single file
-  webpackConfig.module.loaders.push({
-    test: /\.css$/,
-    loader: ExtractTextPlugin.extract("style-loader", "css-loader")
-  });
-
-  webpackConfig.plugins.push(new ExtractTextPlugin("styles.css"));
-}
-
-if (isFirefoxPanel()) {
-  webpackConfig = require("./webpack.config.devtools")(webpackConfig);
-}
-
-// NOTE: This is only needed to fix a bug with chrome devtools' debugger and
-// destructuring params https://github.com/devtools-html/debugger.html/issues/67
-if (getValue("transformParameters")) {
-  webpackConfig.module.loaders.forEach(spec => {
-    if (spec.isJavaScriptLoader) {
-      const idx = spec.loaders.findIndex(loader => loader.includes("babel"));
-      spec.loaders[idx] += "&plugins[]=transform-es2015-parameters";
-    }
-  });
-}
```

Comment:
Yeah, you all are right, we should definitely share it. I'd just like to make sure we do it in a way that's easy to extend, and it's clear where the base config is coming from. I think we're on the same page.